### PR TITLE
Add solution for denied permissions with flagging NEVER ASK AGAIN.

### PR DESCRIPTION
**Pull request by danke77**
If the user denied permissions with flagging NEVER ASK AGAIN, nothing more should happen.
To know if the user denied with flagging NEVER ASK AGAIN we can check again the shouldShowRequestPermissionRationale method in onRequestPermissionsResult when the user did not grant the permission. Then open another dialog explaining the permissions again and directing to the app setting.

### CODE REVIEW

**Comment by danke77** [comment]
I signed it!

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -3,4 +3,7 @@
     <string name="rationale_camera">This app needs access to your camera so you can take pictures.</string>
     <string name="rationale_location_contacts">This app needs access to your location and contacts to know where and who you are.</string>
     <string name="rationale_sms">This app needs access to your sms to read all your great messages.</string>
+    <string name="rationale_ask_again">This app needs to open permissions before using the feature.</string>
+    <string name="setting">SETTING</string>
```
Comment (samtstern) [inline app/src/main/res/values/strings.xml]: nit: replace these two Strings with 'Settings' and 'Cancel'.  Android will take care of capitalizing them when appropriate.

File: app/src/main/java/pub/devrel/easypermissions/sample/MainActivity.java [hunk 2]
```diff
@@ -99,5 +99,8 @@ public void onPermissionsGranted(int requestCode, List<String> perms) {
     @Override
     public void onPermissionsDenied(int requestCode, List<String> perms) {
         Log.d(TAG, "onPermissionsDenied:" + requestCode + ":" + perms.size());
+
+        EasyPermissions.checkDeniedPermissionsNeverAskAgain(this, getString(R.string.rationale_ask_again),
```
Comment (samtstern) [inline app/src/main/java/pub/devrel/easypermissions/sample/MainActivity.java]: Add a comment here explaining what this is doing and that it is optional.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -3,4 +3,7 @@
     <string name="rationale_camera">This app needs access to your camera so you can take pictures.</string>
     <string name="rationale_location_contacts">This app needs access to your location and contacts to know where and who you are.</string>
     <string name="rationale_sms">This app needs access to your sms to read all your great messages.</string>
+    <string name="rationale_ask_again">This app needs to open permissions before using the feature.</string>
```
Comment (samtstern) [inline app/src/main/res/values/strings.xml]: Suggest changing to:
'This app may not work correctly without the requested permissions. Open the app settings screen to modify app permissions'

**Comment by samtstern** [comment]
@danke77 this looks great!  Just a few small comments.

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (danke77) [inline app/src/main/res/values/strings.xml]: Done!

File: app/src/main/java/pub/devrel/easypermissions/sample/MainActivity.java (hunk 2 above)
Comment (danke77) [inline app/src/main/java/pub/devrel/easypermissions/sample/MainActivity.java]: Done!

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (danke77) [inline app/src/main/res/values/strings.xml]: Done!

**Comment by danke77** [comment]
Updated the code based on review comments. Thanks for your time!

**Comment by samtstern** [comment]
@danke77 I have published version `0.1.7` with your changes and updated the README.  Thanks again!

**Pull request merged**